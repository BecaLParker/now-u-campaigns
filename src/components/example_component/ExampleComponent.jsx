import React, { useState } from 'react';
import classes from './exampleComponent.module.scss';
import ExampleChildComponent from './ExampleChildComponent'

const ExampleComponent = () => {
  const [exampleState, setExampleState] = useState(0)
  
  const sayHelloWorld = () => {
    return 'Hello world!'
  }

  const triggerTimeout = () => {
    setTimeout(() => {
      
    }, 10000);
  }

  const handleSubmit = (e) => {
    e.preventDefault()
  }
  
  return (
    <div className={classes.exampleContainer}>
      <h1 className={classes.exampleTitle}>Example Title</h1>
      
      <button id="social-button">Instagram</button>
      
      <div id="example-state">{exampleState}</div>
      
      <a href={classes.exampleLink}>Example Link</a>
      
      <button id='click-to-change-state' onClick={() => setExampleState(1)}>Click Me To Change State</button>

      <ExampleChildComponent />

      <form onSubmit={handleSubmit}>
        <input type="text" value="Enter some information"/>
      </form>

      <button id='trigger-timeout' onClick={triggerTimeout}>Click me to trigger timeout</button>
    </div>
  );
};

export default ExampleComponent;