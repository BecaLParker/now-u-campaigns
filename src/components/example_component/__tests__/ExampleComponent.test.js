import React from 'react'
import { render } from '@testing-library/react'
import ExampleComponent from '../ExampleComponent'
import { shallow, mount } from 'enzyme';

// please see https://jest.io/docs/en for full information on testing
describe('Example app testing for the Example Component', () => {
  
  let wrapper;

  beforeEach(() => {
    // shallow will NOT load child components inside of the one stated,
    // but rather just the component itself.
    // use console.log(wrapper.debug) to see the tree.
    // to see how to load all components including children, use mount
    wrapper = shallow(<ExampleComponent />); 
    // wrapper = mount(<ExampleComponent />); // This will load all children components
    jest.useFakeTimers()
  })

  // Running all pending timers and switching to real timers using Jest
  afterEach(() => {
    jest.runOnlyPendingTimers()
    jest.useRealTimers()
  })

  test("the component renders", () => {
    expect(wrapper.exists()).toBe(true)
  })
  // Testing text
  test("testing the title", () => {
    // console.log(wrapper.debug) use this command to debug
    expect(wrapper.find('h1').text()).toContain("Example Title")
  })

  // Testing a button or ID
  test("render a button with text of `Instagram`", () => {
    expect(wrapper.find('#social-button').text()).toBe('Instagram')
  })

  // Testing state from an item
  test("render the initial value of state in a div", () => {
    expect(wrapper.find('#example-state').text()).toBe("0")
  })

  // Testing that state changes on an event
  test("render the click event of a button and stateChange", () => {
    wrapper.find('#click-to-change-state').simulate('click');
    expect(wrapper.find('#example-state').text()).toBe('1')
  })

  // Testing functions
  test("test hello world function", () => {
    // If you are using CLASS based components, use the syntax below
    // expect(wrapper.instance().sayHelloWorld()).toMatchSnapshot(); <----
    // If you are using Functional Components, test their behaviour as 
    // you wont be able to test their methods.
  })

  // Testing timers
  test("testing that timeout with fake timer", () => {
    wrapper.find("#trigger-timeout").simulate('click');
    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 10000);
  })

  // Testing that preventDefault works
  test("Event is cancelled on form submit", () => {
    let prevented = false;
    wrapper.find("form").simulate('submit', {
      preventDefault: () => {
        prevented = true;
      }
    });
    expect(prevented).toBe(true)
  })


})
