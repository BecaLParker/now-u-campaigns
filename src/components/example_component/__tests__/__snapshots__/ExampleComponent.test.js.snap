// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example app testing for the Example Component test hello world function 1`] = `ShallowWrapper {}`;
